### DISCUSSION TITLE
Upgrade to Vite 6

### DISCUSSION THREAD

**Discussion by rostyslav-nabok** [nrwl/nx#29364] [Feature Requests]
Hey! Can we upgrade `@nx/vite` to [Vite 6](https://vite.dev/blog/announcing-vite6)?
Thank you!

**Comment by Coly010** [comment]
Yes, we will be working on implementing support for Vite 6 and Vitest 3 in the coming weeks

**Reply by YonatanKra** [reply]
Hi. Any news on that one?
With browser testing getting more stability in `vitest 3.X` it is a show stopper ATM.
Is there a temporary workaround?

**Reply by bgriffith** [reply]
> Hi. Any news on that one?
> 
> With browser testing getting more stability in `vitest 3.X` it is a show stopper ATM.
> 
> Is there a temporary workaround?

Looks like it's arriving imminently (20.5.0 release which is currently in RC stage). The PR https://github.com/nrwl/nx/pull/29871

**Reply by YonatanKra** [reply]
Amazing!
Thanks!

**Reply by johnholliday** [reply]
Any ETA on this?

**Reply by rendmath** [reply]
You can manually upgrade `vitest`, `@vitest/coverage-v8` and `@vitest/ui` to 3.x and it seems to work fine (tested on Nx 20.7.1). Unfortunately the Nx migration does not take care of it.

**Comment by Ccry11** [comment]
Hi everyone,
I have upgraded to NX 20.5.0 that supports Vite 6.
I still see the following warning in the console: `The CJS build of Vite's Node API is deprecated. See https://vite.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.`

Will there be a migration to fix this?

**Reply by Ccry11** [reply]
@Coly010 any update on this deprecation warning?

**Reply by rendmath** [reply]
Switching your package.json to type module should solve the CJS version issue (although it might cause other problems).

**Reply by Ccry11** [reply]
In the documentation the first statement is "The CJS build of Vite's Node API is deprecated and will be removed in Vite 6". If we are using Vite 6, how can this work?

**Comment by rendmath** [comment]
The migration upgrades Vite to 6 but does not upgrade Vitest to 3. Which is not a good idea because NPM ends up installing a private 5.x version of Vite under the node_modules of Vitest to satisfy its version dependency.
